extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,122 @@
+                || string.IsNullOrEmpty(blobName)
+                || string.IsNullOrEmpty(containerName))
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var connectionString = connectionName is null ? null : Environment.GetEnvironmentVariable(connectionName);
+            object result = ToTargetType(context.TargetType, connectionString, containerName, blobName);
+
+            if (result is not null)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Success(result));
+            }
+
+            return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+        }
+
+        private object? ToTargetType(Type targetType, string connectionString, string containerName, string blobName) => targetType switch
+        {
+            Type _ when targetType == typeof(String)            => GetBlobString(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Stream)            => GetBlobStream(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Byte[])            => GetBlobBinaryData(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlobClient)        => CreateBlobReference<BlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlockBlobClient)   => CreateBlobReference<BlockBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(PageBlobClient)    => CreateBlobReference<PageBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(AppendBlobClient)  => CreateBlobReference<AppendBlobClient>(connectionString, containerName, blobName),
+            _ => CreateTargetObject(targetType, connectionString, containerName, blobName)
+        };
+
+        private object CreateTargetObject(Type targetType, string connectionString, string containerName, string blobName)
+        {
+            var content = GetBlobString(connectionString, containerName, blobName);
+            MethodInfo deserializeObjectMethod = GetType()
+                                                .GetMethod(nameof(DeserializeTargetObject), BindingFlags.Instance | BindingFlags.NonPublic)
+                                                .MakeGenericMethod(new Type[] { targetType });
+
+            return deserializeObjectMethod.Invoke(this, new object[] { content });
+        }
+
+        private object? DeserializeTargetObject<T>(string content)
+        {
+            return JsonConvert.DeserializeObject<T>(content);
+        }
+
+        private string GetBlobString(string connectionString, string containerName, string blobName)
+        {
+            var client = CreateBlobReference<BlobClient>(connectionString, containerName, blobName);
+            var download = client.DownloadContent();
```

Comment:
We need to be using the async APIs here and everywhere else we're downloading blobs